wsapi.go

```diff
@@ -215,16 +219,22 @@ func (s *Session) heartbeat(wsConn *websocket.Conn, listening <-chan interface{}
 	defer ticker.Stop()
 
 	for {
+		s.RLock()
```

Comment:
This is not writing anything, it's reading a value. Why would i need a write lock?